Marlin/planner.h

```diff
@@ -393,7 +393,7 @@ class Planner {
 
     #if ENABLED(ENSURE_SMOOTH_MOVES)
       static bool long_move() {
-          if (blocks_queued()) {
+          if (blocks_queued() && block_buffer_runtime_us) {
```

Comment:
Ah, ok. The duration of the currently-running block is subtracted before it starts. Good to know.